### PR TITLE
Creating MLGitAPI to allow multiple projects in the same script

### PR DESCRIPTION
**Pull request by GustavoMaciel**
This PR creates a class in the ML-Git API to hold the root path of a project so that every command of the API can run in the context of said project.

This happens because of the decorator that is automatically added to all methods within the class by its Metaclass, the decorator is responsible for: changing the current dir to the project root, executing the command, changing the current dir back to the origin.

The decorator was implemented using the `pathlib.Path` class, which means that we can tell the API the project root by using: a relative path, an absolute path, or the current working directory (CWD) by not passing any path.

An important thing to keep in mind is that the decorator can only work in the class methods that have the parameter self because the decorator needs to access the instance to gather the root_path attribute.

### CODE REVIEW

File: docs/api/api_scripts/basic_flow.ipynb [hunk 1]
```diff
@@ -49,7 +49,23 @@
...
    "source": [
-    "from ml_git import api"
+    "from ml_git.api import MLGitAPI"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Then we must create a new instance"
```
[the remark this answers is not in the document]
Comment (Mun1z) [reply docs/api/api_scripts/basic_flow.ipynb:59]: In this case, it is the instance of the API, we will make it clearer in the text.

File: tests/integration/test_24_api.py [hunk 2]
```diff
@@ -749,3 +749,23 @@ def test_42_graph_order_by_version_number(self):
...
+        config_project_1 = os.path.join(self.tmp_dir, project_1_dir_name, ML_GIT_DIR, 'config.yaml')
+        config_project_2 = os.path.join(self.tmp_dir, project_2_dir_name,  ML_GIT_DIR, 'config.yaml')
+        self.assertFalse(os.path.exists(config_project_1))
+        self.assertFalse(os.path.exists(config_project_2))
+
+        api_1 = api.MLGitAPI(root_path='./{}'.format(project_1_dir_name))
+        api_1.init('repository')
+        self.assertTrue(os.path.exists(config_project_1))
+
+        api_2 = api.MLGitAPI(root_path='./{}'.format(project_2_dir_name))
```
[the remark this answers is not in the document]
Comment (GustavoMaciel) [reply tests/integration/test_24_api.py]: Changed in b76960d

File: tests/integration/test_24_api.py [hunk 3]
```diff
@@ -749,3 +749,23 @@ def test_42_graph_order_by_version_number(self):
...
+        api_2.init('repository')
+        self.assertTrue(os.path.exists(config_project_2))
```
[the remark this answers is not in the document]
Comment (GustavoMaciel) [reply tests/integration/test_24_api.py:780]: Added in b76960d

File: docs/api/api_scripts/basic_flow.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (GustavoMaciel) [reply docs/api/api_scripts/basic_flow.ipynb]: Changed in b76960d

**Review by lucasncabral — APPROVED** [review]

**Review by Mun1z — APPROVED** [review]

File: docs/api/api_scripts/basic_flow.ipynb [hunk 4]
```diff
@@ -49,7 +49,23 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Then we must create a new instance of the API class"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
```
[the remark this answers is not in the document]
Comment (Mun1z) [reply docs/api/api_scripts/basic_flow.ipynb]: Done on last commit: https://github.com/HPInc/ml-git/pull/181/commits/c1ba9c0f53c958371b058c91e4815bf472f41710

**Review by andersonpablosilva — APPROVED** [review]

**Pull request merged**